### PR TITLE
Chrome e2e tests

### PR DESCRIPTION
**Pull request by foresthoffman**
### Describe in detail the problem you had and how this PR fixes it
- [x] `--port` flag
- [x] `--password` flag
- [x] `--allow-http` flag
- [x] `--no-auth` flag
- [x] Can load in successfully.
- [x] Can open file
- [x] Can create file
- [x] Can delete file
- [x] Can install extension
- [x] Can uninstall extension
- [x] Can debug JavaScript file

### Is there an open issue you can link to?
Fixes https://github.com/codercom/code-server/issues/339

### CODE REVIEW

File: packages/app/chrome/test/chrome.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { resolve } from "path";
```
Comment (kylecarbs) [inline packages/app/chrome/test/chrome.test.ts]: `packages/app/chrome` will be disappearing soon. Might be nice to have this in `test/*`

File: packages/app/chrome/test/chrome.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	beforeEach(async () => {
+		page = await browser.newPage();
+	});
+	afterAll(async () => {
+		await browser.close();
+		server.dispose();
+		await server.killProcesses();
+	});
+
+	server.test("should open IDE", async () => {
```
Comment (kylecarbs) [inline packages/app/chrome/test/chrome.test.ts]: I'd use normal `it` here if possible. Will assist in readability.

File: packages/app/chrome/test/chrome.test.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		// page.once("load", async () => {
+		// 	await page.goto(server.url);
+		// 	await page.screenshot({ path: resolve(__dirname, "./screenshot-test.jpg"), fullPage: true });
+		// });
+
+		await page.waitFor(4000);
```
Comment (kylecarbs) [inline packages/app/chrome/test/chrome.test.ts]: I'd create a general function for loading the IDE page.

File: packages/app/chrome/test/chrome.test.ts (hunk 1 above)
Comment (foresthoffman) [reply packages/app/chrome/test/chrome.test.ts]: `packages/app/test`?

Comment (kylecarbs) [reply packages/app/chrome/test/chrome.test.ts]: Nah just `test/`

Comment (kylecarbs) [reply packages/app/chrome/test/chrome.test.ts]: Like at the root of the repo. These are tests for the overall build of the repo afterall.

File: packages/app/chrome/test/chrome.test.ts (hunk 2 above)
Comment (foresthoffman) [reply packages/app/chrome/test/chrome.test.ts]: My thought process is that:
```
server.test("should do a thing", async () => {
    await doThing();
}, 1000);
```
Is prettier than:
```
it("should do a thing", async () => {
    try {
        await doThing();
    } catch (e) {
        // make sure the server closes itself, if the thing doesn't work
        await this.dispose();
        throw e;
    }
}, 1000);
```

Comment (code-asher) [reply packages/app/chrome/test/chrome.test.ts]: Could you dispose the server in `afterEach`?

Comment (foresthoffman) [reply packages/app/chrome/test/chrome.test.ts]: Yes, that would. I've made changes so that's no longer necessary. The tests are using `it` now.

**Comment by code-asher** [comment]
Just had a thought that it might be good to run the tests as Mac, Windows, and Linux clients. Those values are provided by the user agent, could possibly use that? Or import and override the values provided here: https://github.com/microsoft/vscode/blob//src/vs/base/browser/browser.ts#L126

**Comment by foresthoffman** [comment]
@code-asher It is possible to set the user agent for a page with puppeteer: https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagesetuseragentuseragent. I think it uses the base platform to determine the user agent.

**Pull request closed without merging**